scripts/boot_part

```diff
@@ -76,16 +76,25 @@ def set_boot_partition(blkdev, index):
     devnode = blkdev + str(index)
     mntpath = tempfile.mkdtemp()
     try:
-        out = runcmd('sudo mount {0} {1}'.format(devnode, mntpath))
+        out = runcmd(['sudo', 'mount', devnode, mntpath])
         logger.info('mount out={0}'.format(out))
         if out is None: return
         ## Set GRUB bootable
-        out = runcmd('sudo grub-install --boot-directory="{0}" --recheck "{1}"'.format(mntpath, blkdev))
+        out = runcmd(['sudo', 'grub-install', '--boot-directory='+mntpath, "--recheck", blkdev])
         return out is not None
     finally:
         ## Cleanup
-        out = runcmd('sudo fuser -km {0} || sudo umount {0}'.format(mntpath))
-        logger.info('fuser out={0}'.format(out))
+        cmd1 = ['sudo', 'fuser', '-km', mntpath]
+        rc, _ = getstatusoutput_noshell(cmd1)
```

Comment:
>_ [](http://example.com/codeflow?start=12&length=1)

Originally the out1 is also captured and logged, however the original log message is misleading.

You can logging both out1 and out2, no matter success or failure.